src/components/context/QuizContext.tsx

```diff
@@ -201,12 +207,40 @@ export const QuizProvider = ({ children }: Props) => {
         APP_SETTING_NAMES.QUESTION_LIST
       );
 
+      // Get all questions id. To support legacy code, if no question id, the id is used instead.
+      const questionIds = getSettingsByName(
+        settings,
+        APP_SETTING_NAMES.QUESTION
+      ).map((appSetting) => {
+        const qId = appSetting?.data?.questionId;
+        if (qId) {
+          return qId;
+        } else {
+          return appSetting.id;
+        }
+      });
+
       if (newOrderSetting && newOrderSetting.length > 0) {
         const value = newOrderSetting[0] as QuestionListType;
         setOrderSetting(value);
-        setOrder(value?.data?.list ?? []);
+        // Filter out questions that are not well formatted in AppSettings.
+        const filteredOrder =
+          value?.data?.list.filter((id) => questionIds.includes(id)) ?? [];
+        setOrder(filteredOrder);
+
+        // If there are questions, set current idx to the first one.
+        // If it has already set current idx, don't do it again to not reset curr question on order changed.
+        if (filteredOrder.length && !isLoaded) {
+          setCurrentIdx(0);
+        }
+      }
+
+      // if it is first loading, set is loaded to true.
+      if (!isLoaded) {
+        setIsLoaded(true);
       }
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
You sure you want to disable this error? Is there specific variable you want to prevent reloading this useEffect? Maybe it's worth splitting in multiple useEffect?

Reply:
Yes, because I don't want to reload this useEffect when isLoaded change.  I can not split in another useEffect, because I have to listen on settings and I have to set isLoaded and check the value of isLoaded, so I would have to add the eslint.